src/inference/src/dev/core_impl.cpp

```diff
@@ -295,20 +286,76 @@ ov::Parsed ov::parseDeviceNameIntoConfig(const std::string& deviceName,
             OPENVINO_THROW("Device priority / ID mismatch: ",
                            parsed_device_priority,
                            " (from ",
-                           deviceName,
+                           device_name,
                            ") vs ",
                            it->second.as<std::string>(),
                            " (from config)");
         }
     };
 
+    parsed._core_config.set(updated_config);
     // keep batch property only when called from query_supported_property
-    if (!keep_core_property) {
-        clean_batch_properties(updated_device_name, updated_config, ov::hint::allow_auto_batching);
-        clean_batch_properties(updated_device_name, updated_config, ov::auto_batch_timeout);
+    if (!keep_auto_batch_property) {
+        for (const auto& name : auto_batch_properties_names) {
+            clean_batch_properties(updated_device_name, updated_config, name);
+        }
+    }
+    return parsed;
+}
+}  // namespace
+
+bool ov::is_config_applicable(const std::string& user_device_name, const std::string& subprop_device_name) {
```

Comment:
why have you moved this function? 
it will change git history for changes in this function and it will be hard to `git blame`

Reply:
I didn't change it, I think is git interpretation. I can re-order code to make smaller diff

Reply:
Updated, now function is not visible as changed.